Few corrections and replace deprecated functions

npaster opened the pull request.

datafl4sh commented:
@npaster thank you for the PR. I saw that you did some cleanup & refactoring, however it is pretty huge. Could you please give an overview of what you did? Thanks :)

npaster commented:
> @npaster thank you for the PR. I saw that you did some cleanup & refactoring, however it is pretty huge. Could you please give an overview of what you did? Thanks :)

Of course, I modified:
- I replaced mesh_type::scalar_type by mesh_type::coordinate_type everywhere
- I replaced mesh_h by average_diameter
- I deleted all functions which have the tag [deprecated]
- I created a more generic structure for the different behaviors for mechanics
- I corrected few bugs in plasticity behaviors
- I suppressed the conversion between int and size_t in changing the programmation

datafl4sh commented:
OK @npaster, great. Thanks also for taking some time to do refactoring, the thing of `coordinate_type` for example was pretty important to do.
